enhancements/disaster-recovery/automated-cluster-backups.md

```diff
@@ -0,0 +1,324 @@
+  - "https://rancher.com/docs/rancher/v2.x/en/backups/v2.5/configuration/backup-config/"
+  - "https://www.openshift.com/blog/ocp-disaster-recovery-part-1-how-to-create-automated-etcd-backup-in-openshift-4.x"
+---
+
+# Automated Cluster Backups
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Enable the configuration of automated cluster backups.
+
+## Motivation
+
+The current documented procedure for performing a backup of an OpenShift
+cluster is manually initiated. This reduces the likelyhood of a timely backup
+being available in the event of catastrophic cluster failure.
+
+The procedure also requires gaining a root shell on a control plane
+node. Shell access to OpenShift control plane nodes access is generally
+discouraged due to the potential for affecting the reliability of the
+node. Indeed, at least one customer reports having had operator error result
+in a control plane node being taken offline in the process of attempting to
+follow the documented backup procedure.
+
+Finally, OpenShift 4.9 is intended to ship with etcd 3.5, and this will mean
+an upgrade for existing clusters from etcd 3.4. Since etcd will not support
+downgrade from 3.5 to 3.4, downgrade of a OpenShift cluster from 4.9 to 4.8
+will only be possible by restoring from backup. Ensuring support for
+automated backups in the 4.8 edge required to upgrade to 4.9 will increase
+the likelyhood that customers will have cluster backups available in case a
+downgrade is required.
+
+
+### Goals
+
+1. One-time cluster backup can be initiated without requiring a root shell.
+1. Automated backups can be configured as a day 2 operation.
+1. Cluster backups are written to the host filesystem of control plane nodes.
+
+### Non-Goals
+
+1. Backup to cluster (i.e. PVC) or cloud storage (e.g. S3).
+1. Support day 1 configuration of automated backups.
+1. Automate cluster restoration.
+1. Rewrite the existing cluster backup script.
+
+## Proposal
+
+- Add a new API singleton supporting backup configuration
+  - Suggested resource name `backups.config.openshift.io`
+    - Separate backup configuration from the etcd operator in case
+      implementation migrates to a dedicated backup operator.
+  - Fields:
+    - spec.reason `string`
+      - Set to a value different from the most recent reason, will prompt a
+        one-time backup.
+      - Default: `""`
+    - spec.schedule `string`
+      - [Standard cron expression](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#cron-schedule-syntax)
+      - Results in backups scheduled repeatedly to successive control plane
```

Comment:
This is not clear to me.  Can you provide an example of "every two hours" for me?

Also, why a full cron syntax instead of a simple duration?

Reply:
> Also, why a full cron syntax instead of a simple duration?

ok, I see why.  I don't want every 24 hours.  I want everyday at 2am.  Please describe the reasoning in the doc for future me?

Reply:
Done